### DISCUSSION TITLE
How to Implement a Resizable Splitter in Chakra UI?

### DISCUSSION THREAD

**Discussion by D7x7z49** [chakra-ui/chakra-ui#8778] [Q&A] [answered]
Hello Chakra UI Team,

Firstly, thank you for your incredible work on this library—it's been instrumental in my projects!

I’m currently working on a project where I need to implement a resizable splitter, similar to what is often seen in IDEs or admin dashboards, where users can drag a divider between two panes to adjust their sizes.

For example, consider a layout with two adjacent content areas (left and right panes) where a user can click and drag the divider between them to resize the panes. I’m looking for a way to achieve this effect using Chakra UI, ideally with minimal custom CSS or JavaScript, to maintain the simplicity and elegance that Chakra UI provides.

Could you please guide me on how to implement this feature? Is there a recommended approach within the existing Chakra UI framework, or would I need to integrate an external library to accomplish this?

Any examples, suggestions, or insights would be greatly appreciated.

**Comment by mrdotkg** [comment]
@D7x7z49 What did you end up using? There is no feature like that in ChakraUI. 

I like `Golden Layout library`. However, `react-splitter-layout`, `react-split-pane` and `react-resizable` are good options for multi panel, draggable, resizable react components layout library.

**Reply by D7x7z49** [reply]
I implemented it manually, without using any tools. 😢

**Reply by BinarySenseiii** [reply]
https://github.com/bvaughn/react-resizable-panels/tree/main/packages/react-resizable-panels

Why reinvent the wheel when there are already well-tested and well-maintained libraries available? You should check out resizable-panels!!

**Reply by D7x7z49** [reply]
> https://github.com/bvaughn/react-resizable-panels/tree/main/packages/react-resizable-panels
> 
> Why reinvent the wheel when there are already well-tested and well-maintained libraries available? You should check out resizable-panels!!

I think that’s a valid approach! I just feel that adding more features wouldn’t necessarily be a conflict. Chakra can still benefit from reusing existing implementations — my thinking is more about improving convenience and integration.

**Reply by jonahallibone** [reply]
> https://github.com/bvaughn/react-resizable-panels/tree/main/packages/react-resizable-panels
> 
> 
> 
> Why reinvent the wheel when there are already well-tested and well-maintained libraries available? You should check out resizable-panels!!

This library is great if you don't want pixel based sizing in your layout. Unfortunately, this is tough with responsive layouts where you want min/max sizes in pixels or other more fixed layouts.

**Comment by jonahallibone** [comment]
Also would love to see splitter come to v3

**Comment by msimbirsky** [comment]
Looks like a Splitter was added to Ark UI - would love to see Chakra v3 pull this in https://ark-ui.com/docs/components/splitter

**Comment by segunadebayo** [comment, marked as answer]
Splitter shipped. https://chakra-ui.com/docs/components/splitter
